### ISSUE TITLE
Early stoppping for training Request

### ISSUE THREAD

**Issue by tanakataiki** [Unity-Technologies/ml-agents#1718]
![image](https://user-images.githubusercontent.com/37890520/52894921-25a79900-31f5-11e9-800d-0633d01d26e8.png)

I am trainig to mimic walking motion etc yet It fail to train at a certain point suddenly and agent do nothing after that . 

Do you have any idea how to and where to add early stopping in training?
https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/EarlyStopping

I can restore from check points and change learning rate , end learning rate  in model/ppo.py 
batch sizer 4096 
buffer size  40960 
would be enough for walker agent like project.
So It would be greate if someone tell me how to add it.

start at
lr=3e-4 
endlr setting 1e-8
at optimizer
Thanks

**Comment by TashaSkyUp** [comment]
It's hard to tell from just the graphs and a short description. Is this one
of the gym environments?

**Comment by tanakataiki** [comment]
This is python ppo environment not gym 
if any more information is necesssary please tell me.
thanks.

**Comment by awjuliani** [comment]
Hi @tanakataiki 

It is possible to stop training early by pressing `CTL+C` from the training console window.

**Comment by TashaSkyUp** [comment]
Looks like its doing something... hard to tell what.. What are your
hyper-parameters?

**Comment by tanakataiki** [comment]
@awjuliani  Thanks I know that early stoppping is also possible manually but I want to stop automaticallly from script since the training takes usually 2-4days. and I cant decide the right step to stop. But I think the epoch which value estimate suddenly decrease or sudden increase of policy loss would be a good trigger to stop.

@TashaSkyUp 
Well, this problem would happen in any projects since training to much would end up in getting local minimum if the task is difficult

    learning_rate: 3.0e-4
    beta: 5.0e-3
    normalize: true
    num_epoch: 3
    time_horizon: 1000
    batch_size: 4096
    buffer_size: 40960
    gamma: 0.95
    lambd: 0.95
    max_steps: 4e6
    summary_freq: 10000
    num_layers: 2
    hidden_units: 1024.

my hyperparameters by the way.
 The project I do is 
https://qiita.com/tanakataiki/items/94c7778341a40438cc99#runningin-the-middle-of-training
here.

Thanks.

**Comment by awjuliani** [comment]
Hi @tanakataiki 

We don't support this, but are happy to accept PRs for it.

**Comment by surfnerd** [comment]
Hello.  Thank you for your discussion.  We are closing it now due to inactivity. Please reopen it if you want to further discuss this topic.
